cosmo_tester/test_suites/snapshots/multi_tenant_snapshot_test.py

```diff
@@ -13,141 +13,182 @@
+
+
+@pytest.fixture(
+        scope='module',
+        params=['master', '4.1', '4.0.1'])
+def cluster_multitenant(request, cfy, ssh_key, module_tmpdir, attributes,
+                        logger, install_dev_tools=True):
+    cluster = _cluster(request, cfy, ssh_key, module_tmpdir, attributes,
+                       logger, 2, install_dev_tools)
     yield cluster
-
     cluster.destroy()
 
 
-@pytest.fixture(autouse=True)
-def _multi_tenant_hello_world_example(cluster, attributes, logger):
-    manager0 = cluster.managers[0]
-
-    for tenant in TENANTS:
-        manager0.client.tenants.create(tenant)
-        manager0.upload_plugin('openstack_centos_core', tenant_name=tenant)
-        update_client_tenant(manager0.client, tenant)
-        deploy_helloworld(attributes, logger, manager0)
-
-    update_client_tenant(manager0.client, DEFAULT_TENANT_NAME)
-
-    yield
-
-    if not manager0.deleted:
-        try:
-            for tenant in TENANTS:
-                update_client_tenant(manager0.client, tenant)
-                logger.info('Cleaning up hello_world_example deployment...')
-                execution = manager0.client.executions.start(
-                    deployment_id,
-                    'uninstall',
-                    parameters={'ignore_failure': True}
-                    )
-                wait_for_execution(
-                    manager0.client,
-                    execution,
-                    logger,
-                    )
-        except Exception as e:
-            logger.error('Error on test cleanup: %s', e)
-
-
-def test_restore_snapshot_and_agents_upgrade(
-        cfy, cluster, attributes, logger, tmpdir):
-    manager0 = cluster.managers[0]
-    manager1 = cluster.managers[1]
-
-    snapshot_id = str(uuid.uuid4())
-
-    logger.info('Creating snapshot on old manager..')
-    manager0.client.snapshots.create(snapshot_id, False, True, False)
-    assert_snapshot_created(manager0, snapshot_id, attributes)
-
-    local_snapshot_path = str(tmpdir / 'snapshot.zip')
-
-    logger.info('Downloading snapshot from old manager..')
-    manager0.client.snapshots.list()
-    manager0.client.snapshots.download(snapshot_id, local_snapshot_path)
-
-    manager1.use()
-    logger.info('Uploading snapshot to latest manager..')
-    snapshot = manager1.client.snapshots.upload(local_snapshot_path,
-                                                snapshot_id)
-    logger.info('Uploaded snapshot:%s%s',
-                os.linesep,
-                json.dumps(snapshot, indent=2))
-
-    cfy.snapshots.list()
-
-    logger.info('Restoring snapshot on latest manager..')
-    restore_execution = manager1.client.snapshots.restore(snapshot_id)
-    logger.info('Snapshot restore execution:%s%s',
-                os.linesep,
-                json.dumps(restore_execution, indent=2))
-
-    cfy.executions.list(['--include-system-workflows'])
+def create_tenant_secrets(manager, tenants, logger):
```

Comment:
Add an explanation for `create_tenant_secrets ` and `check_tenant_secrets `. Why is it here, what does it do, etc.